frontend/test/unit/specs/stores/feature-flag-store.spec.js

```diff
@@ -143,16 +143,19 @@ describe("Feature flag store", () => {
     ${"staging"}    | ${"off"}
     ${"production"} | ${"off"}
   `(
-    "returns $expectedState for $environment",
+    "returns $featureState for $environment",
     ({ environment, featureState }) => {
+      const initialValue = process.env.DEPLOYMENT_ENV
+      process.env.DEPLOYMENT_ENV = environment
       const featureFlagStore = useFeatureFlagStore()
-      featureFlagStore.$nuxt.$config.deploymentEnv = environment
+
       expect(featureFlagStore.featureState("feat_env_specific")).toEqual(
         featureState
       )
       expect(featureFlagStore.isOn("feat_env_specific")).toEqual(
         featureState === "on"
       )
+      process.env.DEPLOYMENT_ENV = initialValue
```

Comment:
Here as well, regarding use of the before/after hooks to ensure proper cleanup in all cases.